Source/ACE/Entity/Player.cs

```diff
@@ -243,20 +243,59 @@ public Player(Session session) : base(ObjectType.Creature, session.CharacterRequ
             Session.WorldSession.EnqueueSend(setTurbineChatChannels, general, trade, lfg, roleplay);
         }
 
+        /// <summary>
+        /// Raise the available XP by a specified amount
+        /// </summary>
+        /// <param name="amount">A unsigned long containing the desired XP amount to raise</param>
         public void GrantXp(ulong amount)
         {
             character.GrantXp(amount);
+            //TODO: Ask for an Answer: Does CheckForLevelup() go before or after the xp message?
+            CheckForLevelup();
             var xpTotalUpdate = new GameMessagePrivateUpdatePropertyInt64(Session, PropertyInt64.TotalExperience, character.TotalExperience);
             var xpAvailUpdate = new GameMessagePrivateUpdatePropertyInt64(Session, PropertyInt64.AvailableExperience, character.AvailableExperience);
             var message = new GameMessageSystemChat($"{amount} experience granted.", ChatMessageType.Broadcast);
             Session.WorldSession.EnqueueSend(xpTotalUpdate, xpAvailUpdate, message);
         }
 
+        /// <summary>
+        /// Determines if the player has advanced a level
+        /// </summary>
+        /// <remarks>
+        /// Known issues:
+        ///         1. Have not tested up to max level, so there isn't any special text or fireworks for max level yet.
+        /// </remarks>
         private void CheckForLevelup()
         {
+            // Question: Where do *we* call CheckForLevelup()? : 
+            //      From within the player.cs file, the options might be:
+            //           GrantXp()
+            //      From outside of the player.cs file, we may call CheckForLevelup() durring? :
+            //           XP Updates?
+
             var chart = DatabaseManager.Charts.GetLevelingXpChart();
+            var currentLevel = character.Level;
+            int maxLevel = chart.Levels.Count;
 
-            // TODO: implement.  just stubbing for now, will implement later.
+            if (character.Level == maxLevel) return;
+
+            // this will check against the xp chart to see if the xp is greater then the last level
+            // increases until the correct level is found
+            while (chart.Levels[Convert.ToInt32(character.Level)].TotalXp <= character.TotalExperience)

```

Comment:
For each level that we have experience to reach, we will advance. This is also the location that should advance skill credits in the future. If maxLevel is achieved, then we will break.